training-react/src/app/components/User/index.js

```diff
@@ -34,42 +32,19 @@ class User extends Component {
     const userSession = user ? this.props.user : JSON.parse(LocalStoreService.getItem(USER_SESSION));
     const { isSettingUser } = this.state;
 
-    let infoUser = <UserInfo userSession={userSession} />;
-    if (isSettingUser) {
-      infoUser = (
-        <UserEdit userSession={userSession} setUser={setUser} editUser={this.editUser} {...this.props} />
-      );
-    }
+    const infoUserSection = isSettingUser ? (
```

Comment:
i think it would be better to pass `isSettingUser` prop to the layout component and let it decide what to render
```jsx
<UserDetailsWithLoading
     isLoading={!user}
     user={user}
     isSettingUser={isSettingUser}
     editUser={this.editUser}
/>

// layout.js
function UserDetails({ user, userSection, editUser }) {
  return (
    ...
    // change this
    // {userSection}
    // for this
    { isSettingUser ? (
        <UserEdit userSession={userSession} setUser={setUser} editUser={this.editUser} {...this.props} />
      ) : (
        <UserInfo userSession={userSession} />
    )}
    ...
}
```